.mk/build.mk

```diff
@@ -96,7 +96,9 @@ docker-push: $(addsuffix -push,$(addprefix docker-,$(images)));
 
 .PHONY: docker-login
 docker-login:
-	@echo "${DOCKER_PASSWORD}" | docker login -u "${DOCKER_USERNAME}" --password-stdin
+	@if [ -n "${DOCKER_PASSWORD}" ]; then \
+		echo "${DOCKER_PASSWORD}" | docker login -u "${DOCKER_USERNAME}" --password-stdin ;\
+	fi
```

Comment:
For diagnostic purpose, it might be good to add 'else' clause with echoing something like 'docker-login: bypassing authorization'. Otherwise it might take some time to debug a situation when DOCKER_PASSWORD is not set by mistake.

Reply:
updated